bpf/bpf_host.c

```diff
@@ -1266,10 +1241,7 @@ int cil_to_host(struct __ctx_buff *ctx)
 	bool traced = false;
 	__u32 src_id = 0;
 
-	if ((magic & MARK_MAGIC_HOST_MASK) == MARK_MAGIC_ENCRYPT) {
-		ctx->mark = magic; /* CB_ENCRYPT_MAGIC */
-		src_id = ctx_load_meta(ctx, CB_ENCRYPT_IDENTITY);
```

Comment:
For reviewers: On a previous backport of this change, Julian commented:
> Just to note that in v1.13 the wireguard implementation is different, and `from-container` uses `set_encrypt_mark()`. So I hope our reasoning why we were able to remove this section in `main` also applies to v1.13...

Martynas confirmed that this code isn't used in the case of WireGuard on v1.13 and older. Instead, traffic marked for WireGuard encryption is sent to cilium_wg0 and then traverses `to-host` without a mark.

Reply:
I must have not been paying sufficient attention during that review. In the `from-container` path, Wireguard uses `set_encrypt_mark()` - so the `MARK_MAGIC_ENCRYPT` ends up in `skb->mark`. But here we obtained the `magic` value from the `skb->cb` (note the `ctx_load_meta()`).

So unless there's some magic at play that transfers the `skb->cb` to the `skb->mark`, there's no way this could have even conflicted. Sorry for burning your cycles on this :(.

Reply:
No worries at all. Better have unnecessary checks than missing something.